src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -548,31 +562,73 @@ internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration
             }
             hostConfig.UseServiceBus(serviceBusConfig);
 
-            // Apply Tracing configuration
+            // Apply Tracing/Logging configuration
             configSection = (JObject)config["tracing"];
-            if (configSection != null && configSection.TryGetValue("consoleLevel", out value))
+            if (configSection != null)
             {
-                TraceLevel consoleLevel;
-                if (Enum.TryParse<TraceLevel>((string)value, true, out consoleLevel))
+                if (configSection.TryGetValue("consoleLevel", out value))
+                {
+                    TraceLevel consoleLevel;
+                    if (Enum.TryParse<TraceLevel>((string)value, true, out consoleLevel))
+                    {
+                        hostConfig.Tracing.ConsoleLevel = consoleLevel;
+                    }
+                }
+
+                if (configSection.TryGetValue("fileLoggingEnabled", out value))
                 {
-                    hostConfig.Tracing.ConsoleLevel = consoleLevel;
+                    scriptConfig.FileLoggingEnabled = (bool)value;
                 }
             }
 
+            // Set up a host level TraceMonitor that will receive notificaition
+            // of ALL errors that occur. This allows us to inspect/log errors.
+            var traceMonitor = new TraceMonitor()
```

Comment:
This is the TraceMonitor extension stuff. It allows us to receive all errors that occur, and we route them below in HandleHostError